doc/specifications/proposed/content_download/content_download.md

```diff
@@ -0,0 +1,60 @@
+# Content download
+
+> Author: bd@ez.no
+
+## Summary
+This feature covers downloading of Content binary files over HTTP.
+
+It covers:
+- generation of proper links from Twig as well as PHP,
+- permissions checking (the user must be allowed to read the Content)
+- streaming of the file over HTTP
+
+## Components
+
+### Twig helper
+
+> Name: ez_download_path
+> Usage: `ez_download_path( content, 'file' )`
+
+The helper will be used to generate a download link for a given Field/Content.
+
+### Controller action + route
+
+#### Route
+
+Path: `/content/download/{contentId}/{fieldIdentifier}/{fileName}`
+
+Example: ``/content/download/68/file/My file.pdf
+
+Query parameters
+
+- version (optional)
+
+  The version number the file must be downloaded for. Requires the versionview permission.
+  If not specified, the published version is used.
```

Comment:
May I suggest something slightly different?

What if it just work like a hint, not a version selector, _but it is always there_, so like`?latestVersionNo=2`.
Means client side cache can be safely made to last long, and if for some reason someone is hitting this url against the backend server, we can do a Permanent redirect (and cache the response) to the currently known latests version. This might produce several redirects if one is cached and new version has been created after the cache was created, but that should be ok.


For version view url I would suggest this is slightly different and always check version view rights.
But @dpobel might have some input on if that would work or not.